functionalTest/api/foo/fooStubTest.js

```diff
@@ -1,107 +1,90 @@
+            return api.post('/imposters', request).then(() => tcp.send('request', port)
+            ).then(response => {
                 assert.strictEqual(response.toString(), 'first');
-
                 return tcp.send('request', port);
-            }).then(function (response) {
+            }).then(response => {
                 assert.strictEqual(response.toString(), 'second');
-
                 return tcp.send('request', port);
-            }).then(function (response) {
+            }).then(response => {
                 assert.strictEqual(response.toString(), 'first');
-
                 return tcp.send('request', port);
-            }).then(function (response) {
+            }).then(response => {
                 assert.strictEqual(response.toString(), 'second');
-            }).finally(function () {
-                return api.del('/imposters');
-            });
+            }).finally(() => api.del('/imposters'));
         });
 
-        promiseIt('should only return stubbed response if matches complex predicate', function () {
-            var stub = {
+        promiseIt('should only return stubbed response if matches complex predicate', () => {
+            const stub = {
                     responses: [{ is: { data: 'MATCH' } }],
                     predicates: [
                         { equals: { data: 'test' } },
                         { startsWith: { data: 'te' } }
                     ]
                 },
-                request = { protocol: 'foo', port: port, stubs: [stub] };
+                request = { protocol: 'foo', port, stubs: [stub] };
 
-            return api.post('/imposters', request).then(function (response) {
+            return api.post('/imposters', request).then(response => {
                 assert.strictEqual(response.statusCode, 201, JSON.stringify(response.body));
                 return tcp.send('not test', port, 100);
-            }).then(function (response) {
+            }).then(response => {
                 assert.strictEqual(response.toString(), 'foo');
-
                 return tcp.send('test', port, 100);
-            }).then(function (response) {
+            }).then(response => {
                 assert.strictEqual(response.toString(), 'MATCH');
-            }).finally(function () {
-                return api.del('/imposters');
-            });
+            }).finally(() => api.del('/imposters'));
         });
 
-        promiseIt('should allow proxy stubs', function () {
-            var originServerPort = port + 1,
+        promiseIt('should allow proxy stubs', () => {
+            const originServerPort = port + 1,
                 originServerStub = { responses: [{ is: { data: 'PROXIED' } }] },
                 originServerRequest = {
                     protocol: 'foo',
                     port: originServerPort,
                     stubs: [originServerStub],
-                    name: this.name + ' ORIGIN'
+                    name: `${requestName} ORIGIN`
```

Comment:
This is an example where, if we can't get the name of the test ("should allow proxy stubs") or it's hard, we should just remove requestName. As is, it's more confusing than just leaving it as "ORIGIN"